File: Java/Practice/src/InsertionSort.java

public class InsertionSort {
    public static void main(String[] args) {

        int[] num = {3,2,2,2,2, 4, 5, 8, 9, 10, 20, 21, 15, 17, 13, 12, 14, 9, 2, 4, 12, 14, 19, 21, 23, 26, 28, 31, 51};
        int[] dupedNum = new int[num.length];
        int temp = 0, duplicate = 0;
        // boolean isSorted;

        for (int i = 1; i < num.length; i++) {
            int j = i;
            while (j > 0 && num[j - 1] > num[j]) {
                temp = num[j];
                num[j] = num[j - 1];
                num[j - 1] = temp;
                j--;

            }

        }
        System.out.print("Sorted Array: ");

        for (int i = 0; i < num.length; i++) {
            if (i > 0 && num[i] == num[i - 1]) {
                if (dupedNum[num[i]] == 0) {
                    dupedNum[num[i]] = num[i];
                    duplicate++;
                }
            }
            System.out.printf("%d ", num[i]);
        }
        System.out.print("\n\n");
        System.out.print("Duplicated Numbers: ");
        for (int i = 0; i < num.length; i++) {
            if (dupedNum[i] != 0) {
                System.out.printf("%d ", dupedNum[i]);
            }
//            System.out.printf("%d ", dupedNum[i]);

        }
        System.out.printf("\nUnique Duped Num: %d", duplicate);
    }
}

File: C/Practice/InsertionSort.c
#include <stdio.h>

int main()
{
    int num[] = {2,8,3,2,4,5,6,7,9,10,23,1,10,10,2,8,3,2,4,5,6,7,9,10,23,1,10,10};
    int temp = 0, duplicate =0;
    int dupeNum[sizeof(num) / sizeof(num[0])] = {0};
    
    // Insertion Sort Alogrithm =======================================
    for(int i =1; i < sizeof(num) / sizeof(num[0]); i++){
        int j =i;
        while(j>0 && num[j-1] > num[j]){

            temp = num[j];
            num[j] = num[j-1];
            num[j-1] = temp;
            j--;
        }
    }
    // Searching Through the Array for Duplicates and Putting them in dupeNum ARRAY =======================================

    printf("Sorted Array: ");
    for(int i =0; i<sizeof(num)/ sizeof(num[0]); i++){
        // int j =i;
        if(i >0 && num[i]==num[i-1]){
            
            if(dupeNum[num[i]] == 0){
                dupeNum[num[i]] = num[i-1];
                duplicate++;
            }
        }
        printf("%d ", num[i]);
    }
    printf("\n\n");

    // Displaying the duplicate number and the number of unique dupe numbers =======================================

    printf("Duped Numbers: ");  
    for(int i =1; i<sizeof(dupeNum)/ sizeof(dupeNum[0]); i++){
        if(dupeNum[i] != 0){
            printf("%d ", dupeNum[i]);
        }
        // printf("%d ", dupeNum[i]);
    }
    
    printf("\nDuplicate Numbers: %d", duplicate);
    

    return 0;
}